### PR TITLE
Rename CdfFileEntry to CdcEntry

### LINKED ISSUE #17183: Rename CdfFileEntry to CdcEntry
CdfFileEntry is not in line with https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-cdc-file
should be renamed to CdcEntry

### PR DESCRIPTION
**Pull request by marcinsbd**
## Description

Fixes #17183

### CODE REVIEW

**Review by homar — APPROVED** [review]

**Review by ebyhr — COMMENTED** [review]
There're still many "cdf" usages in the connector. Did you leave it intentionally? What's the criteria to use cdf or cdc in the code?

**Comment by ebyhr** [comment]
> Fix issue: https://github.com/trinodb/trino/issues/17183

This comment doesn't close the issue automatically. Please use the template in https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword. e.g. `Fixes #17183`

**Comment by homar** [comment]
> There're still many "cdf" usages in the connector. Did you leave it intentionally? What's the criteria to use cdf or cdc in the code?

So my understanding is that the name of the feature is CDF (change data feed) and databricks also uses this name for the feature, however the name of the entry in the log is for some reason cdc. The idea was to only rename parts that directly touches that cdc log entry entity.
Does it make sese? WDYT ?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java [hunk 1]
```diff
@@ -1608,7 +1608,7 @@ public void finishMerge(ConnectorSession session, ConnectorMergeTableHandle merg
         }
     }
 
-    private static void appendCdfFileEntries(
+    private static void appendCdcEntries(
             TransactionLogWriter transactionLogWriter,
             List<DataFileInfo> cdfFilesInfos,
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1613]: The method name was changed to use "cdc". I would rename to `cdcFilesInfos` including the method caller's argument.

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java]: The method caller still uses `cdfFiles`.

**Pull request merged**